From a683764be9f12938c63cfe0465caf41975f0ee74 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <j@w1.fi>
Date: Wed, 18 Sep 2024 10:59:25 +0530
Subject: [PATCH] UPSTREAM: hostapd: Clear rejected groups list on completing
 authentication The rejected groups list is valid only during each individual
 SAE authentication instance and it should not be maintained between separate
 instances. In particular, it should not be maintained when roaming to another
 AP since the APs might use different configuration for the allowed SAE
 groups.

Tag: SAE
Patch-work: None
Patch-dependency: None

Signed-off-by: Jouni Malinen <j@w1.fi>
Signed-off-by: Arunpandi Kannan <quic_arunpand@quicinc.com>
---
 wpa_supplicant/sme.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/wpa_supplicant/sme.c b/wpa_supplicant/sme.c
index a47a3c1..c479745 100644
--- a/wpa_supplicant/sme.c
+++ b/wpa_supplicant/sme.c
@@ -1984,6 +1984,7 @@ static int sme_sae_auth(struct wpa_supplicant *wpa_s, u16 auth_transaction,
 
 		wpa_s->sme.sae.state = SAE_ACCEPTED;
 		sae_clear_temp_data(&wpa_s->sme.sae);
+		wpa_s_clear_sae_rejected(wpa_s);
 
 		if (external) {
 			/* Report success to driver */
-- 
2.34.1

